Core/include/Acts/Seeding/SpacePointGrid.ipp

```diff
@@ -10,57 +10,78 @@
 
 #include <memory>
 
+#include <iostream>
+
+using AxisScalar = Acts::Vector3::Scalar;
+
 template <typename SpacePoint>
 std::unique_ptr<Acts::SpacePointGrid<SpacePoint>>
 Acts::SpacePointGridCreator::createGrid(
     const Acts::SpacePointGridConfig& config) {
-  int phiBins;
-  // for no magnetic field, create 100 phi-bins
-  if (config.bFieldInZ == 0) {
-    phiBins = 100;
-  } else {
-    // calculate circle intersections of helix and max detector radius
-    float minHelixRadius =
-        config.minPt / (300. * std::abs(config.bFieldInZ));  // in mm
-    float maxR2 = config.rMax * config.rMax;
-    float xOuter = maxR2 / (2 * minHelixRadius);
-    float yOuter = std::sqrt(maxR2 - xOuter * xOuter);
-    float outerAngle = std::atan(xOuter / yOuter);
-    // intersection of helix and max detector radius minus maximum R distance
-    // from middle SP to top SP
-    float innerAngle = 0;
-    if (config.rMax > config.deltaRMax) {
-      float innerCircleR2 =
-          (config.rMax - config.deltaRMax) * (config.rMax - config.deltaRMax);
-      float xInner = innerCircleR2 / (2 * minHelixRadius);
-      float yInner = std::sqrt(innerCircleR2 - xInner * xInner);
-      innerAngle = std::atan(xInner / yInner);
-    }
-
-    // FIXME: phibin size must include max impact parameters
-    // divide 2pi by angle delta to get number of phi-bins
-    // size is always 2pi even for regions of interest
-    phiBins = std::floor(2 * M_PI / (outerAngle - innerAngle));
+ 
+  // calculate circle intersections of helix and max detector radius
+  float minHelixRadius = config.minPt / (300. * config.bFieldInZ);  // in mm -> R[mm] =pT[GeV] / (3·10−4×B[T]) =  pT[MeV] / (300 *Bz[kT])
```

Comment:
`config.bFieldInZ` can be 0, in that case, `minHelixRadius` is infinite. This is why the check and an arbitrary default there.